src/components/expanded-state/token-history/TokenHistory.js

```diff
@@ -0,0 +1,342 @@
+import { Contract } from '@ethersproject/contracts';
+import MaskedView from '@react-native-community/masked-view';
+import React, { useCallback, useEffect, useState } from 'react';
+import { FlatList } from 'react-native';
+import { ButtonPressAnimation } from '../../animations';
+import EdgeFade from './EdgeFade';
+import { useTheme } from '@rainbow-me/context/ThemeContext';
+import {
+  AccentColorProvider,
+  Box,
+  Inline,
+  Inset,
+  Stack,
+  Text,
+} from '@rainbow-me/design-system';
+import {
+  apiGetNftSemiFungibility,
+  apiGetNftTransactionHistoryForEventType,
+} from '@rainbow-me/handlers/opensea-api';
+import { web3Provider } from '@rainbow-me/handlers/web3';
+import { formatAssetForDisplay } from '@rainbow-me/helpers';
+import { getHumanReadableDate } from '@rainbow-me/helpers/transactions';
+import { useAccountProfile, useAccountSettings } from '@rainbow-me/hooks';
+import { useNavigation } from '@rainbow-me/navigation';
+import NetworkTypes from '@rainbow-me/networkTypes';
+import {
+  ENS_NFT_CONTRACT_ADDRESS,
+  NULL_ETH_ADDRESS,
+  REVERSE_RECORDS_MAINNET_ADDRESS,
+  reverseRecordsABI,
+} from '@rainbow-me/references';
+import Routes from '@rainbow-me/routes';
+import { handleSignificantDecimals } from '@rainbow-me/utilities';
+import { abbreviations } from '@rainbow-me/utils';
+import { EventTypes, PaymentTokens } from '@rainbow-me/utils/tokenHistoryUtils';
+import logger from 'logger';
+
+const TokenHistory = ({ contractAddress, tokenID, accentColor }) => {
+  const [tokenHistory, setTokenHistory] = useState([]);
+  const { colors } = useTheme();
+  const { accountAddress } = useAccountProfile();
+  const { network } = useAccountSettings();
+  const { navigate } = useNavigation();
+
+  const networkPrefix = network === NetworkTypes.mainnet ? '' : `${network}-`;
+
+  useEffect(() => {
+    if (!accountAddress) {
+      logger.sentry('NFT transaction history: wallet address not found');
+      setTokenHistory(null);
+      return;
+    }
+    const processRawEvents = async (contractAddress, rawEvents) => {
```

Comment:
I don't think this whole thing should be part of useEffect, maybe extract it into its own hook?